app/controllers/course/video/topics_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+class Course::Video::TopicsController < Course::Video::Controller
+  load_and_authorize_resource :topic, class: Course::Video::Topic.name
+  before_action :load_existing_topic, only: :create
+
+  include Course::Discussion::PostsConcern
+  skip_load_and_authorize_resource :post, except: :create
+
+  def index
+    @topics = @video.topics.includes(posts: :children).order(:timestamp)
+    @topics = @topics.reject { |topic| topic.posts.empty? }
+    @posts = @topics.map(&:posts).inject(Course::Discussion::Post.none, :+)
+  end
+
+  def create
+    result = @topic.class.transaction do
```

Comment:
Should we catch the possibility that `@topic` might be `nil`?

Reply:
Hmm, the @topic here is first populated with a new topic by `load_and_authorize_resource`. Only when a timestamp is found do we replace it with an existing topic. So it should never be null?

Reply:
So just to clarify, this create method actually handles the creation of a post (which can be a new topic, or part of an existing topic)?

Reply:
Yup, so timestamps are tagged to topics, and the frontend always sends a timestamp for the creation of posts. But if it happens to coincide with an existing topic down to a difference of 1 second, then a new post is nested under it instead.

The exception is when a parent post is specified, then it's always nested under the parent post.

Reply:
Ok I think that was what threw me off a little since. When do write the controller tests, do test for this logic as well.

Reply:
Alright! Will do